### DISCUSSION TITLE
gpt4-pdf-chatbot-langchain working like  https://www.chatpdf.com/

### DISCUSSION THREAD

**Discussion by beralc** [mayooear/ai-pdf-chatbot-langchain#171] [Q&A]
Is it possible to make it work like this: https://www.chatpdf.com/? This app combines both the text from the PDF and chatgpt. e.g. if you ask who Elvis is, it will tell you even if it's not in the book, and yoru are able to do other things around the PDF, for example, write reading comprehension exercises.

**Comment by abrainc** [comment]
Yes, you can do this easily by editing the prompts in makechain.ts. The prompts tell the model not to use any information outside the documents.

**Reply by beralc** [reply]
I tried that. Basically I'd like for the app to read the pdf and be able to make reading comprehension exercises. Like try/false questions etc. but even if I put it in the prompt, it won't do it. And it won't use chatgpt if I ask a question that isn't in the pdf. Even if I write in the prompt "use chatgpt if you don't know the answer". This is what chatpdf kind of does. Any help would be greatly appreciated.

**Reply by abrainc** [reply]
There is also a second prompt, editing, that might help. As far as I can tell, the model is not using outside knowledge because of the prompts; I changed mine to include phases like "Some questions might require to use critical thinking skills to answer." Other than that, I am not too sure.

**Reply by abrainc** [reply]
see https://github.com/mayooear/gpt4-pdf-chatbot-langchain/discussions/144#discussioncomment-5560646 they had a similar issue

**Reply by beralc** [reply]
Yeah. I tried that, but it won't do it. Any ideas on how to solve this are appreciated. As I said my goal is to make reading comprehension exercises for my students. So, upload the pdf and then ask the bot to create multiple choice questions or reading comprehension questions etc. I don't know if this is possible.

**Comment by felipeotarola** [comment]
Rewrite the prompt actually works. 
I asked Chatgtp to write a prompt for me that first checks my document and if its not in the document but related information it gets info from it own.

**Reply by disturbed-mystic1** [reply]
Could you share the prompt used.
